src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Services/DataAccess/DtoMappers/PaymentDtoMapper.cs

```diff
@@ -6,10 +6,10 @@ namespace TicketManager.PaymentService.Services.DataAccess.DtoMappers;
 
 public class PaymentDtoMapper
 {
-    public readonly static Expression<Func<Payment, PaymentDto>> ToDtoMapper = o => new PaymentDto()
+    public readonly static Expression<Func<Payment, PaymentDto>> ToDtoMapper = p => new PaymentDto()
     {
-        Id = o.Id,
-        PaymentStatus = (PaymentStatusDto)o.PaymentStatus,
-        DateCreated = o.DateCreated
+        Id = p.Id,
+        PaymentStatus = p.HasExpired? PaymentStatusDto.Expired: (PaymentStatusDto)p.PaymentStatus,
+        DateCreated = p.DateCreated
     };    
```

Comment:
It's weird that here ef doesn't have problem with `HasExpired`, but has problem with it in validator.